### PR TITLE
Add `ClientAvatar` & `ClientListItem` components

### PR DESCRIPTION
**Pull request by ohaibbq**
## Description of the change

These components will be used in the Compliant Reporting selection flow

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes [#XXXX]

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/ClientAvatar/ClientAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+const AvatarElement = styled.div(
+  ({ primary, secondary, offset }: AvatarElementProps) => `
+  background-color: ${primary};
+  background-image: radial-gradient(ellipse at ${offset}% ${offset}%, ${primary}, transparent),
+          radial-gradient(ellipse at ${100 - offset}% ${
+    100 - offset
+  }%, ${secondary}, transparent);
+
```
Comment (Catacola) [inline src/core/ClientAvatar/ClientAvatar.tsx:39]: Juan is going to love you

File: src/core/ClientAvatar/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import ClientAvatar from "./ClientAvatar";
+
+export { ClientAvatar };
```
Comment (macfarlandian) [inline src/core/ClientAvatar/index.ts:3]: `export * from "./ClientAvatar"`?

File: src/core/ClientAvatar/ClientAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  offset: number;
+}
+
+const AvatarElement = styled.div(
+  ({ primary, secondary, offset }: AvatarElementProps) => `
+  background-color: ${primary};
+  background-image: radial-gradient(ellipse at ${offset}% ${offset}%, ${primary}, transparent),
+          radial-gradient(ellipse at ${100 - offset}% ${
+    100 - offset
+  }%, ${secondary}, transparent);
```
Comment (macfarlandian) [inline src/core/ClientAvatar/ClientAvatar.tsx:38]: i think this is probably fine for now but in his mocks Juan was using this package called [Boring Avatars](https://github.com/boringdesigners/boring-avatars)

Comment (ohaibbq) [reply src/core/ClientAvatar/ClientAvatar.tsx:38]: 🤣 I was having fun with it, but am more than happy to switch over

File: src/core/ClientListItem/ClientListItem.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+interface ClientListItemProps {
+  client: Client;
+}
+
+const ClientListItemElement = styled.div`
+  align-items: center;
+  display: grid;
+  grid-template-columns: ${rem(40)} 1fr;
+  margin-bottom: ${rem(spacing.sm)};
+`;
```
Comment (Catacola) [inline src/core/ClientListItem/ClientListItem.tsx:34]: I'm not sure what the mocks call for, but this element feels like it could use a `:hover` section.

**Review by Catacola — APPROVED** [review]
Whoohoo!

**Review by macfarlandian — APPROVED** [review]

**Pull request merged**